### PR TITLE
Documentation for the back-channel communication 

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- Add a new tutorial regarding back-channel communication.

**Related issue(s)**

https://github.com/kyma-project/cli/issues/824

### CODE REVIEW

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 1]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+
+<div tabs name="signature" group="function-specification">
+<details>
+<summary label="Node.js">
+Node.js
+</summary>
+
+| Function Name | Arguments | Description |
+|---------------|-----------|-------------|
+| setResponseHeader | key, value | Set a header to the response object based on the given `key` and the `value` |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:124]:
```suggestion
| **setResponseHeader** | key, value | Sets a header to the response object based on the given key and the value |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 2]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+| setResponseContentType | type | Set the `ContentType` header to the response object based on the given `type` |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:125]:
```suggestion
| **setResponseContentType** | type | Sets the `ContentType` header to the response object based on the given type |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 3]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+| setResponseStatus | status | Set the response statuc based on the given `status` |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:126]:
```suggestion
| **setResponseStatus** | status | Sets the response status based on the given status |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 4]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+| publishCloudEvent | event | Publish an CloudEvent to the publisher service based on the given CloudEvent object |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:127]:
```suggestion
| **publishCloudEvent** | event | Publishes a CloudEvent on the publisher service based on the given CloudEvent object |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 5]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+| buildResponseCloudEvent | id, type, data | Build an CloudEvent object based on the request CloudEvent object and given arguments |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:128]:
```suggestion
| **buildResponseCloudEvent** | id, type, data | Builds a CloudEvent object based on the request CloudEvent object and given arguments |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 6]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+
+</details>
+<details>
+<summary label="Python">
+Python
+</summary>
+
+| Function | Arguments | Description |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:136]:
```suggestion
| Function name | Arguments | Description |
```or "Method name" (depending on the answer to the question from one of the above comments.)

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 7]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+|----------|-----------|-------------|
+| publishCloudEvent | event | Publish an CloudEvent to the publisher service based on the given CloudEvent object |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:138]:
```suggestion
| **publishCloudEvent** | event | Publishes a CloudEvent on the publisher service based on the given CloudEvent object |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 8]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
...
+| buildResponseCloudEvent | id, type, data | Build an CloudEvent object based on the request CloudEvent object and given arguments |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:139]:
```suggestion
| **buildResponseCloudEvent** | id, type, data | Builds a CloudEvent object based on the request CloudEvent object and given arguments |
```

**Review by mmitoraj — CHANGES_REQUESTED** [review]
Please see my comments below. Thank you.

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 9]
```diff
@@ -109,6 +109,38 @@ See the detailed descriptions of these fields:
 | **data** | Either JSON or a string, depending on the request type. Read more about [Buffer](https://nodejs.org/api/buffer.html) in Node.js and [bytes literals](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals) in Python. |
 | **extensions** | JSON object that can contain event payload, a Function's incoming request, or an outgoing response |
 
+### Event object SDK
+
+The `event` object is extended by methods making things easier. You can use every method by writing `event.Function_Name(Arguments...)`.
```
[the remark this answers is not in the document]
Comment (pPrecel) [reply docs/05-technical-reference/svls-08-function-specification.md:114]: Definitely `ARGUMENTS`. The number of arguments is given in the `Arguments` section in the table and yup, my fault, the method is the same thing as the function. I will change it to the `function`

**Review by kwiatekus — DISMISSED** [review]

File: docs/03-tutorials/eventing/evnt-01-setup-in-cluster-eventing.md [hunk 10]
```diff
@@ -47,3 +47,5 @@ EOF
     -H "Content-Type: application/cloudevents+json" \
     "http://eventing-event-publisher-proxy.kyma-system/publish"
 ```
+
+> **NOTE:** If you want to use a Function to publish a CloudEvent look at the [Event Object SDK specification](../../05-technical-reference/svls-08-function-specification.md).
```
Comment (mmitoraj) [inline docs/03-tutorials/eventing/evnt-01-setup-in-cluster-eventing.md:51]:
```suggestion
> **NOTE:** If you want to use a Function to publish a CloudEvent, see the [Event object SDK specification](../../05-technical-reference/svls-08-function-specification.md#event-object-sdk).
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 11]
```diff
@@ -109,6 +109,39 @@ See the detailed descriptions of these fields:
...
+
+<div tabs name="signature" group="function-specification">
+<details>
+<summary label="Node.js">
+Node.js
+</summary>
+
+| Function Name | Arguments | Description |
+|---------------|-----------|-------------|
+| **setResponseHeader** | key, value | Sets a header to the response object based on the given key and the value |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:124]:
```suggestion
| **setResponseHeader** | key, value | Sets a header to the response object based on the given key and value |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 12]
```diff
@@ -109,6 +109,39 @@ See the detailed descriptions of these fields:
...
+| **setResponseContentType** | type | Sets the `ContentType` header to the response object based on the given type |
+| **setResponseStatus** | status | Sets the response status based on the given status |
+| **publishCloudEvent** | event | Publishes a CloudEvent on the publisher service based on the given CloudEvent object |
+| **buildResponseCloudEvent** | id, type, data | Builds a CloudEvent object based on the request CloudEvent object and given arguments |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:128]:
```suggestion
| **buildResponseCloudEvent** | id, type, data | Builds a CloudEvent object based on the request CloudEvent object and the given arguments |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 13]
```diff
@@ -109,6 +109,39 @@ See the detailed descriptions of these fields:
...
+
+</details>
+<details>
+<summary label="Python">
+Python
+</summary>
+
+| Function name | Arguments | Description |
+```or "Method name" (depending on the answer to the question from one of the above comments.)
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:137]:
```suggestion

```
Remove the line and fix the toggle, please.

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 14]
```diff
@@ -109,6 +109,39 @@ See the detailed descriptions of these fields:
...
+
+The `event` object is extended by functions making some operations easier. You can use every function by providing `event.{FUNCTION_NAME(ARGUMENTS...)}`.
+
+<div tabs name="signature" group="function-specification">
+<details>
+<summary label="Node.js">
+Node.js
+</summary>
+
+| Function Name | Arguments | Description |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:122]:
```suggestion
| Function name | Arguments | Description |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 15]
```diff
@@ -109,6 +109,39 @@ See the detailed descriptions of these fields:
...
+<details>
+<summary label="Python">
+Python
+</summary>
+
+| Function name | Arguments | Description |
+```or "Method name" (depending on the answer to the question from one of the above comments.)
+|----------|-----------|-------------|
+| **publishCloudEvent** | event | Publishes a CloudEvent on the publisher service based on the given CloudEvent object |
+| **buildResponseCloudEvent** | id, type, data | Builds a CloudEvent object based on the request CloudEvent object and given arguments |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:140]:
```suggestion
| **buildResponseCloudEvent** | id, type, data | Builds a CloudEvent object based on the request CloudEvent object and the given arguments |
```

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 16]
```diff
@@ -109,6 +109,39 @@ See the detailed descriptions of these fields:
...
+
+</details>
+</div>
+
 ### Context object
 
 The `context` object contains information about the Function invocation, such as runtime details, execution timeout, or memory limits.
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:147]:
```suggestion
The `context` object contains information about the Function's invocation, such as runtime details, execution timeout, or memory limits.
```

**Review by mmitoraj — CHANGES_REQUESTED** [review]
Hi, I added a few more comments. Please make sure that the toggle displays correctly. Thank you.

**Review by kwiatekus — DISMISSED** [review]

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 17]
```diff
@@ -109,9 +109,41 @@ See the detailed descriptions of these fields:
 | **data** | Either JSON or a string, depending on the request type. Read more about [Buffer](https://nodejs.org/api/buffer.html) in Node.js and [bytes literals](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals) in Python. |
 | **extensions** | JSON object that can contain event payload, a Function's incoming request, or an outgoing response |
 
+### Event object SDK
+
+The `event` object is extended by functions making some operations easier. You can use every function by providing `event.{FUNCTION_NAME(ARGUMENTS...)}`.
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:114]:
```suggestion
The `event` object is extended by Functions making some operations easier. You can use every Function by providing the following: `event.{FUNCTION_NAME(ARGUMENTS...)}`.
```
Is `event` an object or an argument? See the **Signature** section -> 

"Functions must accept two arguments that are passed to the Function handler:
- `event`
- `context`"

File: docs/05-technical-reference/svls-08-function-specification.md [hunk 18]
```diff
@@ -109,9 +109,41 @@ See the detailed descriptions of these fields:
...
+| **publishCloudEvent** | event | Publishes a CloudEvent on the publisher service based on the given CloudEvent object |
+| **buildResponseCloudEvent** | id, type, data | Builds a CloudEvent object based on the `request` CloudEvent object and the given arguments |
+
+</details>
+<details>
+<summary label="Python">
+Python
+</summary>
+
+| Function name | Arguments | Description |
```
Comment (mmitoraj) [inline docs/05-technical-reference/svls-08-function-specification.md:136]:
```suggestion
| Method name | Arguments | Description |
```

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**